### ISSUE TITLE
read_table behaviour with more names than columns

### ISSUE THREAD

**Issue by Phlya** [open2c/bioframe#105]
When reading a 3-columns file using `bioframe.read_table` with schema `bed4`, bioframe doesn't error, and simply returns a 4-column bedframe with NaN names. Perhaps it should error? At least cooltools seems to expect it! https://github.com/open2c/cooltools/blob/f33c512ae6ecf8eab69a4d752e70350a293e9564/cooltools/lib/io.py#L360

See discussion here: https://open2c.slack.com/archives/CTFSERWAE/p1642087289008500

**Comment by sergpolly** [comment]
I wonder if something has changes on pandas side ?! e.g.
```python
# faking a 4-column text file
yyy = 'chr1\t1000\t3000\treg1\nchr2\t5000\t10000\treg2\n'
ccc = io.StringIO(yyy)
# requesting more names than there are columns ...
pd.read_csv(ccc, sep="\t", header=None,names=list("abcdefg"))
```
I think the above was failing before ... It does not fail now (`pandas 1.3.2`) ...
I couldnt find anything relevant in the changelog https://pandas.pydata.org/docs/whatsnew/index.html

But I'm not 100% positive at this point...

Regardless we need to update cooltools to conform this 100%

**Comment by gfudenberg** [comment]
one possibility:

```python
def read_table(filepath_or, schema=None, schema_is_strict=False, **kwargs):
```
if any of columns are all NAs:
- schema_is_strict=False throws a warning 
- schema_is_strict=True throws an error

could also check if number of columns read < number of columns in table

**Comment by gfudenberg** [comment]
addressed in  #106
